examples/apps/generic_client/generic_client.py

```diff
@@ -19,465 +19,267 @@
 
-    return True, wo_params
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO)
+TCF_HOME = os.environ.get("TCF_HOME", "../../../")
 
 
-def _create_work_order_receipt(wo_receipt, wo_params,
-                               client_private_key, jrpc_req_id):
-    # Create a work order receipt object using WorkOrderReceiptRequest class.
-    # This function will send a WorkOrderReceiptCreate JSON RPC request.
-    wo_request = json.loads(wo_params.to_jrpc_string(jrpc_req_id))
-    wo_receipt_request_obj = WorkOrderReceiptRequest()
-    wo_create_receipt = wo_receipt_request_obj.create_receipt(
-        wo_request,
-        ReceiptCreateStatus.PENDING.value,
-        client_private_key
-    )
-    logger.info("Work order create receipt request : {} \n \n ".format(
-        json.dumps(wo_create_receipt, indent=4)
-    ))
-    # Submit work order create receipt jrpc request
-    wo_receipt_resp = wo_receipt.work_order_receipt_create(
-        wo_create_receipt["workOrderId"],
-        wo_create_receipt["workerServiceId"],
-        wo_create_receipt["workerId"],
-        wo_create_receipt["requesterId"],
-        wo_create_receipt["receiptCreateStatus"],
-        wo_create_receipt["workOrderRequestHash"],
-        wo_create_receipt["requesterGeneratedNonce"],
-        wo_create_receipt["requesterSignature"],
-        wo_create_receipt["signatureRules"],
-        wo_create_receipt["receiptVerificationKey"],
-        jrpc_req_id
-    )
-    logger.info("Work order create receipt response : {} \n \n ".format(
-        wo_receipt_resp
-    ))
+class GenericClient():
+    """
+    Generic client class to test end to end flow
+    for both direct model and proxy model
+    """
 
+    def __init__(self, args):
+        # Parse command line arguments
+        options = self._parse_command_line(args)
+        # Read config params
+        if options.config:
+            self._config = self._parse_config_file(options.config)
+        else:
+            self._config = self._parse_config_file(
+                TCF_HOME +
+                "/sdk/avalon_sdk/tcf_connector.toml")
+        if self._config is None:
+            logging.error("\n Error in parsing config file: {}\n")
+            sys.exit(-1)
+        if options.blockchain:
+            self._config['blockchain']['type'] = options.blockchain
+            self._blockchain = options.blockchain
+        else:
+            self._blockchain = self._config['blockchain']['type']
+
+        self._is_direct_mode = False
+        # Mode
+        self._mode = options.mode
+
+        # Http JSON RPC listener uri
+        self._uri = options.uri
+        if self._uri:
+            self._config["tcf"]["json_rpc_uri"] = self._uri
+
+        # Address of smart contract
+        self._address = options.address
+        if self._address:
+            if self._mode == "listing":
+                self._config["ethereum"]["direct_registry_contract_address"] =\
```

Comment:
"direct_registry_contract_address" should be read based on blockchain type right. ? This is to read the uri from registry (blockchain) in case of direct model if uri is not passed in command line.

Reply:
Currently it is supported for Ethereum only.